test/test_torch.py

```diff
@@ -11360,8 +11360,14 @@ def run_test(device, dtype):
                                                   [2., 1.]]],
                                                 dtype=dtype,
                                                 device=device)
+            expected_unique_dim1_bool = torch.tensor([[[False, True], [True, True]], 
```

Comment:
@mruberry the code here is kind of awkward and it would be good if dtype genericity had a good way to handle this.

Reply:
Handle what, in particular? That the constructor for bool tensors is distinct?

Reply:
Err, yes I guess? I'm not too sure what to do, I just know the code here looks bad.